scripts/docs_update_release.py

```diff
@@ -0,0 +1,216 @@
+idf_version = environ.get('ESP_IDF_VERSION', '')                         # e.g. '4.4.7'
+ide_version = environ.get('ESPRESSIF_IDE_VERSION', '')                   # e.g. '2.13.69'
+online_installer_version = environ.get('ONLINE_INSTALLER_VERSION', '')   # e.g. '2.25'
+
+
+if not idf_version:
+    raise SystemExit("ERROR: IDF version is not provided")
+
+# cast IDF version to list
+match = re.match(r'^(\d+)\.(\d+)(?:\.(\d+))?$', idf_version)
+if match:
+    idf_version:list = [match.group(1),  match.group(2), match.group(3) if match.group(3) else None]
+
+print(f"IDF version: {idf_version}")
+
+new_idf_version = f"{idf_version[0]}.{idf_version[1]}{f'.{idf_version[2]}' if idf_version[2] else ''}"
+
+
+def _resolve_installer_type() -> str:
+    """Resolve the type of the installer
+        and return the string of new entry for the index.html
+    """
+    new_entry_ide = f"""            <div class="download-button">
+                    <form method="get" action="https://dl.espressif.com/dl/idf-installer/espressif-ide-setup-{ide_version}-with-esp-idf-{new_idf_version}.exe">
+                        <button class="button-espressif-ide">
+                            <i class="fa fa-download" aria-hidden="true"></i>
+                            <div>Espressif-IDE {ide_version} with ESP-IDF v{new_idf_version} - Offline Installer</div>
+                            <div>Windows 10, 11</div>
+                            <div>Size: {installer_size}</div>
+                        </button>
+                    </form>
+                </div>"""
+
+    new_entry_offline = f"""            <div class="download-button">
+                    <form method="get" action="https://dl.espressif.com/dl/idf-installer/esp-idf-tools-setup-offline-{new_idf_version}.exe">
+                        <button class="button-offline">
+                            <i class="fa fa-download" aria-hidden="true"></i>
+                            <div>ESP-IDF v{new_idf_version} - Offline Installer</div>
+                            <div>Windows 10, 11</div>
+                            <div>Size: {installer_size}</div>
+                        </button>
+                    </form>
+                  </div>"""
+
+    new_entry_online = f"""            <div class="download-button">
+                    <form class="download-form" method="get" action="https://dl.espressif.com/dl/idf-installer/esp-idf-tools-setup-online-{online_installer_version}.exe">
+                        <button class="button-online">
+                            <i class="fa fa-download" aria-hidden="true"></i>
+                            <div>Universal Online Installer {online_installer_version}</div>
+                            <div>Windows 10, 11</div>
+                            <div>Size: {installer_size}</div>
+                        </button>
+                    </form>
+                </div>"""
+    
+    if installer_type == 'espressif-ide':
```

Comment:
This is acquired from the runner environment at the beginning of the script:

    installer_type: str = environ.get('INSTALLER_TYPE', '')
    
There is no option for this value to be other than `espressif-ide, offline, online` (list of "hard" defined options in workflow) also it corresponds with the strings (names) used for tags.

So I think an enum is not necessary because we are comparing the string environmental value with the corresponding string value.
**But you are right, this should be better commented to explain this fact, I will do this.** Thank you.